pkg/common-controller/snapshot_controller.go

```diff
@@ -1099,6 +1127,19 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.Vo
 		if err != nil {
 			return nil, newControllerUpdateError(utils.SnapshotKey(snapshot), err.Error())
 		}
+
+		// If this is the first time we're setting creation time, we know snapshotting has finished
+		if snapshotObj.Status != nil && snapshotObj.Status.CreationTime == nil &&
```

Comment:
snapshotObj.Status can be nil
also, can use newStatus directly(it should not be nil) instead of checking snapshotClone.Status != nil